flax/linen/module.py

```diff
@@ -1000,6 +1007,17 @@ def __post_init__(self) -> None:
     else:
       raise ValueError('parent must be None, Module or Scope')
 
+    # eagerly bind submodules if scope is available
+    if self.scope is not None:
+      for field in dataclasses.fields(self):
+        if (
+          field.name not in ('parent', 'name')
+          and field.name in self.__dict__ # ignore fields that have not been set
+        ):
+          value = getattr(self, field.name)
+          if _tree_has_modules(value):
```

Comment:
Could you tell me again why we're doing `_tree_has_modules` when `_register_submodules` implements a recursion over submodules?  is that missing something?